apps/rush-lib/src/logic/ProjectWatcher.ts

```diff
@@ -54,29 +54,38 @@ export class ProjectWatcher {
    * Will return immediately the first time it is invoked, since no state has been recorded.
    * If no change is currently present, watches the source tree of all selected projects for file changes.
    */
-  public async waitForChange(): Promise<IProjectChangeResult> {
+  public async waitForChange(onWatchingFiles?: () => void): Promise<IProjectChangeResult> {
     const initialChangeResult: IProjectChangeResult = this._computeChanged();
     // Ensure that the new state is recorded so that we don't loop infinitely
     this._commitChanges(initialChangeResult.state);
     if (initialChangeResult.changedProjects.size) {
       return initialChangeResult;
     }
 
-    const watcher: import('chokidar').FSWatcher = new chokidar.FSWatcher({
-      persistent: true,
-      cwd: Path.convertToSlashes(this._rushConfiguration.rushJsonFolder),
-      followSymlinks: false,
-      ignoreInitial: true,
-      ignored: /(?:^|[\\\/])node_modules/g,
-      disableGlobbing: true,
-      interval: 1000
-    });
-
-    // Only watch for changes in the requested project folders
+    const previousState: PackageChangeAnalyzer = initialChangeResult.state;
+    const repoRoot: string = Path.convertToSlashes(this._rushConfiguration.rushJsonFolder);
+
+    const pathsToWatch: Set<string> = new Set();
+
+    const isRecursiveSupported: boolean = os.platform() === 'win32';
```

Comment:
Oh does this mean "is recursive folder watch supported"" Maybe we can clear up the name.